cypress-io/cypress
# feat: scaffold react18 projects in component setup

- Closes #23033

### User facing changelog
React 18 projects will now be scaffolded with the `cypress/react18` mount registration during component setup

### Additional details
Refactored the `mountModule` framework property to be async, allowing the detection of the installed react version to determine `cypress/react` vs `cypress/react18`. React is the only framework that needs the detection, but I've refactored the other frameworks so they have the same function signature

### Steps to test
Create a project using React 18, for example `npm init vite` -> choose React

Open project with local Cypress, verify `support/component.js` contains `import { mount } from 'cypress/react18'`

### How has the user experience changed?

https://user-images.githubusercontent.com/25158820/183974906-51679798-bee6-4f1e-920a-b771c92c56bf.mov

### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
